plugin/src/test/java/org/opensearch/ml/action/prediction/PredictionITTests.java

```diff
@@ -5,107 +5,84 @@
-import java.util.Collection;
-import java.util.Collections;
-import java.util.concurrent.ExecutionException;
+import static org.opensearch.ml.utils.TestData.IRIS_DATA_SIZE;
 
 import org.junit.Before;
-import org.junit.Ignore;
-import org.opensearch.ResourceNotFoundException;
+import org.junit.Rule;
+import org.junit.rules.ExpectedException;
 import org.opensearch.action.ActionFuture;
 import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.ml.action.MLCommonsIntegTestCase;
+import org.opensearch.ml.common.dataframe.DataFrame;
+import org.opensearch.ml.common.dataset.DataFrameInputDataset;
 import org.opensearch.ml.common.dataset.MLInputDataset;
 import org.opensearch.ml.common.dataset.SearchQueryInputDataset;
 import org.opensearch.ml.common.parameter.FunctionName;
 import org.opensearch.ml.common.parameter.MLInput;
+import org.opensearch.ml.common.parameter.MLModel;
+import org.opensearch.ml.common.parameter.MLPredictionOutput;
 import org.opensearch.ml.common.transport.MLTaskResponse;
 import org.opensearch.ml.common.transport.prediction.MLPredictionTaskAction;
 import org.opensearch.ml.common.transport.prediction.MLPredictionTaskRequest;
-import org.opensearch.ml.plugin.MachineLearningPlugin;
-import org.opensearch.plugins.Plugin;
-import org.opensearch.search.builder.SearchSourceBuilder;
 import org.opensearch.test.OpenSearchIntegTestCase;
 
-@OpenSearchIntegTestCase.ClusterScope(transportClientRatio = 0.9)
-@Ignore("Test cases in this class are flaky, something is off with waitModelAvailable(taskId) method."
-    + " This issue will be tracked in an issue and will be fixed later")
-public class PredictionITTests extends OpenSearchIntegTestCase {
-    private String taskId;
-
-    @Before
-    public void initTestingData() throws ExecutionException, InterruptedException {
-        generateMLTestingData();
-
-        SearchSourceBuilder searchSourceBuilder = generateSearchSourceBuilder();
-        MLInputDataset inputDataset = new SearchQueryInputDataset(Collections.singletonList(TESTING_INDEX_NAME), searchSourceBuilder);
-        taskId = trainModel(inputDataset);
-        waitModelAvailable(taskId);
-    }
-
-    @Override
-    protected Collection<Class<? extends Plugin>> nodePlugins() {
-        return Collections.singletonList(MachineLearningPlugin.class);
-    }
-
-    @Override
-    protected Collection<Class<? extends Plugin>> transportClientPlugins() {
-        return Collections.singletonList(MachineLearningPlugin.class);
-    }
+import com.google.common.collect.ImmutableList;
 
-    public void testTestingData() throws ExecutionException, InterruptedException {
-        verifyGeneratedTestingData(TESTING_DATA);
-        waitModelAvailable(taskId);
-    }
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.SUITE, numDataNodes = 2)
+public class PredictionITTests extends MLCommonsIntegTestCase {
+    private String irisIndexName;
+    private String modelId;
```

Comment:
I think these IT test should be in lieu of the UTs? though they don't show up in the jacoco test reports.

Reply:
Let's keep both UT and IT. Jacoco may not cover the REST API integration test.